core/src/Kokkos_CopyViews.hpp

```diff
@@ -1254,6 +1254,80 @@ struct ViewRemap<DstType, SrcType, ExecSpace, 8> {
   }
 };
 
+template <typename T>
+bool is_zero_byte(const T& t) {
+  using comparison_type = std::conditional_t<
+      sizeof(T) % sizeof(long long int) == 0, long long int,
+      std::conditional_t<
+          sizeof(T) % sizeof(long int) == 0, long int,
+          std::conditional_t<
+              sizeof(T) % sizeof(int) == 0, int,
+              std::conditional_t<sizeof(T) % sizeof(short int) == 0, short int,
+                                 char>>>>;
+  const auto* const ptr = reinterpret_cast<const comparison_type*>(&t);
+  for (std::size_t i = 0; i < sizeof(T) / sizeof(comparison_type); ++i)
+    if (ptr[i] != 0) return false;
+  return true;
+}
+
+template <typename ExecutionSpace, class DT, class... DP>
+inline void plain_memcpy(
+    const ExecutionSpace& exec_space, const View<DT, DP...>& dst,
+    typename ViewTraits<DT, DP...>::const_value_type& value) {
+  using ViewType     = View<DT, DP...>;
+  using ViewTypeFlat = Kokkos::View<
+      typename ViewType::value_type*, Kokkos::LayoutRight,
+      Kokkos::Device<typename ViewType::execution_space,
+                     typename std::conditional<ViewType::Rank == 0,
+                                               typename ViewType::memory_space,
+                                               Kokkos::AnonymousSpace>::type>,
+      Kokkos::MemoryTraits<0>>;
+
+  ViewTypeFlat dst_flat(dst.data(), dst.size());
+  if (dst.span() < static_cast<size_t>(std::numeric_limits<int>::max())) {
+    Kokkos::Impl::ViewFill<ViewTypeFlat, Kokkos::LayoutRight, ExecutionSpace,
+                           ViewTypeFlat::Rank, int>(dst_flat, value,
+                                                    exec_space);
+  } else
+    Kokkos::Impl::ViewFill<ViewTypeFlat, Kokkos::LayoutRight, ExecutionSpace,
+                           ViewTypeFlat::Rank, int64_t>(dst_flat, value,
+                                                        exec_space);
+}
+
+template <typename ExecutionSpace, class DT, class... DP>
+struct ZeroMemset {
+  static void execute(const ExecutionSpace& exec_space,
+                      const View<DT, DP...>& dst,
+                      typename ViewTraits<DT, DP...>::const_value_type& value) {
+    plain_memcpy(exec_space, dst, value);
+  }
+
+  static void execute(const View<DT, DP...>& dst,
+                      typename ViewTraits<DT, DP...>::const_value_type& value) {
+    plain_memcpy(ExecutionSpace(), dst, value);
+  }
+};
```

Comment:
> Please justify the departure from the style used in DeepCopy, i.e. why did you decide to declare static member functions rather than have the logic in the constructors?

We can use constructors.

> Also why is ExecutionSpace a template parameter instead of being deduced?

There were some problems with `CudaUVM` where the three-argument case might use an `ExecutionSpace` different from the one provided by `View`.

> Why do you pass value if you know it is zero?

The general template calls `plain_memcpy` which needs the value. In general, I can't construct (and don't want to construct) an all zero-byte object on the fly so I prefer to just pass `value` to `ZeroMemset` and ignore it in the specializations.